core/src/main/java/io/parsingdata/metal/data/DataExpressionSource.java

```diff
@@ -45,17 +48,17 @@ public DataExpressionSource(final ValueExpression dataExpression, final int inde
     protected byte[] getData(final long offset, final int size) throws IOException {
         final ImmutableList<Optional<Value>> results = dataExpression.eval(graph, encoding);
         if (results.size <= index) { throw new IllegalStateException("ValueExpression dataExpression yields " + results.size + " result(s) (expected at least " + (index + 1) + ")."); }
-        final byte[] inputData = getValueAtIndex(results, index, 0).get().getValue();
+        final byte[] inputData = getValueAtIndex(results, index, 0).computeResult().get().getValue();
```

Comment:
This is a personal thing, but I prefer to create a method with ''logical/expected" parameters, which then passes extra necessary parameters to the computing method, e.g. in this case, you only call ```getValueAtIndex(results, index)```, even for private methods. (also, this method assumes that the given index is valid with respect to the list size?)